starknet_devnet/blueprints/gateway.py

```diff
@@ -21,23 +21,32 @@ async def add_transaction():
     """Endpoint for accepting DEPLOY and INVOKE_FUNCTION transactions."""
 
     transaction = validate_transaction(request.data)
-    tx_type = transaction.tx_type.name
```

Comment:
I moved the dict construction from the end to the top of the function. The dict is now populated step by step.